test/run_test.py

```diff
@@ -132,6 +138,15 @@ def print_to_stderr(message):
 
 
 def run_test(executable, test_module, test_directory, options, *extra_unittest_args):
+    # Adding the root working directory, test_directory, to Python
+    # module search path, sys.path,
+    # because if the test_module is loacated in a subdirectory,
+    # only that subdirectory is in sys.path by default,
+    # `import common_utils` would fail in this case.
+    os.environ['PYTHONPATH'] = os.pathsep.join(
+        [path for path in [os.environ.get('PYTHONPATH', ''), test_directory] if path]
+    )
+
```

Comment:
I think this is silly and you should just keep all the tests in the top level directory. If you do this, will running tests directly from the directory (without using `run_test.py`) work? I don't think so! And now you've just made it that much harder for people to run a specific test with, e.g., gdb attached.

Reply:
@pietern  Agree? Let's put them flatly?